datapipelines/src/main/kotlin/io/customer/datapipelines/plugins/AutomaticActivityScreenTrackingPlugin.kt

```diff
@@ -0,0 +1,62 @@
+package io.customer.datapipelines.plugins
+
+import android.app.Activity
+import android.content.pm.PackageManager
+import com.segment.analytics.kotlin.android.plugins.AndroidLifecycle
+import com.segment.analytics.kotlin.core.Analytics
+import com.segment.analytics.kotlin.core.platform.Plugin
+
+/**
+ * Optional interface for activities that should be tracked using automated screen tracking.
+ */
+interface TrackableScreen {
+    /**
+     * Retrieve the name that should be used for tracking the screen. This name
+     * should be unique for each screen.
+     *
+     * @return name for tracking the screen, or null if the screen shouldn't be tracked.
+     */
+    fun getScreenName(): String?
+}
+
+/**
+ * Plugin that automatically tracks screens via ActivityLifecycleCallbacks `onActivityStarted` event of an activity.
+ */
+class AutomaticActivityScreenTrackingPlugin : Plugin, AndroidLifecycle {
+
+    override val type: Plugin.Type = Plugin.Type.Utility
+    override lateinit var analytics: Analytics
+
+    override fun onActivityStarted(activity: Activity?) {
+        val packageManager = activity?.packageManager
+        try {
+            val screenName: String? = if (activity is TrackableScreen) {
+                // TrackableScreen takes precedence over manifest label
+                activity.getScreenName()
+            } else {
+                val info = packageManager?.getActivityInfo(
+                    activity.componentName,
+                    PackageManager.GET_META_DATA
+                )
+                val activityLabel = info?.loadLabel(packageManager)
+
+                activity?.let {
+                    activityLabel.toString().ifEmpty {
+                        activity::class.java.simpleName.getScreenNameFromActivity()
+                    }
+                }
+            }
+            // If screen name is null, we do not track the screen
+            screenName?.let {
```

Comment:
Should we check for blank here too?

Reply:
it might not be needed but doesnt hurt to add it